xwayland/selection/incoming.c

```diff
@@ -61,7 +61,9 @@ static void xwm_write_property(struct wlr_xwm_selection_transfer *transfer,
 	transfer->property_start = 0;
 	transfer->property_reply = reply;
 
-	xwm_data_source_write(transfer->wl_client_fd, WL_EVENT_WRITABLE, transfer);
+	if (!xwm_data_source_write(transfer->wl_client_fd, WL_EVENT_WRITABLE, transfer)) {
```

Comment:
This is maybe a little confusing, because `xwm_data_source_write` doesn't return a `bool` indicating success/error. Instead, it returns an int which indicates whether we need to re-check the FD. Maybe we could split that up in a variable to make it more explicit? e.g.

```c
bool needs_recheck = xwm_data_source_write(…);
```

Reply:
Good idea, done.